{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import gensim\n",
    "from gensim.models import Word2Vec\n",
    "from gensim.models.callbacks import CallbackAny2Vec\n",
    "import warnings \n",
    "from sklearn.cluster import KMeans\n",
    "from sklearn.cluster import AgglomerativeClustering\n",
    "from sklearn.cluster import ward_tree\n",
    "from sklearn.cluster import Birch \n",
    "from sklearn.cluster import DBSCAN\n",
    "import os\n",
    "warnings.filterwarnings(action = 'ignore') "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/shunling/Desktop/CapstoneProjects/NaturalLanguageProcessingGPCR/'"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "currentdir = os.getcwd()\n",
    "def datadir(file):\n",
    "    return currentdir+'/'+file\n",
    "datadir('')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "class callback(CallbackAny2Vec):\n",
    "    def __init__(self):\n",
    "        self.epoch = 0\n",
    "        self.loss_s = 0\n",
    "        \n",
    "    def on_epoch_end(self,model):\n",
    "            \n",
    "        if self.epoch % 1 == 0:\n",
    "            loss_e = int(model.get_latest_training_loss())\n",
    "            loss = loss_e - self.loss_s\n",
    "            self.loss_s = loss_e\n",
    "            print('Loss after epoch {}: {}'.format(self.epoch, loss))\n",
    "        self.epoch += 1 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "model = Word2Vec.load('Models/m300_50_5_5.model')\n",
    "vocab = model.wv.vocab.keys()\n",
    "word_vecs = [model.wv[word] for word in vocab]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "from datetime import datetime\n",
    "\n",
    "from scipy.spatial import distance\n",
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-11-04 15:44:42.722296 :  0.026670044\n",
      "2019-11-04 17:28:56.396974 :  0.02801668\n",
      "2019-11-04 18:01:31.751550 :  0.028978229\n",
      "2019-11-04 18:35:11.205471 :  0.029265454\n",
      "2019-11-04 19:11:05.824285 :  0.030010302\n",
      "2019-11-04 19:52:22.570462 :  0.031986773\n",
      "2019-11-04 20:29:44.552433 :  0.031823047\n",
      "2019-11-04 21:07:59.560140 :  0.031575613\n"
     ]
    }
   ],
   "source": [
    "# Using the documentation for the `silhouette_score` function above, \n",
    "# construct a series of silhouette plots like the ones in the article linked above.\n",
    "from sklearn.metrics import silhouette_score\n",
    "SSE = []\n",
    "K = range(10,50,5)\n",
    "for k in K:\n",
    "    kmeanModel = KMeans(n_clusters=k)\n",
    "    \n",
    "    score = silhouette_score(word_vecs, kmeanModel.fit_predict(word_vecs))\n",
    "    print(datetime.now(),': ',score)\n",
    "    SSE.append(score)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYAAAAD4CAYAAADlwTGnAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjEsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy8QZhcZAAAgAElEQVR4nO3deXhU9d3+8feHBBLWxISwBgjIIgEBNYJb645YF2q1FVsFrS1aazfbWrWPdXlaW22fWtvaBRVF9OdSl4p7VWy1VpFE1rDITsIOISEQsn9+f+RAY0xkgCRnJnO/risXM2e+M7nnAOeec+Y7Z8zdERGR+NMu7AAiIhIOFYCISJxSAYiIxCkVgIhInFIBiIjEqcSwAxyM7t27e1ZWVtgxRERiSl5e3nZ3z2i4PKYKICsri9zc3LBjiIjEFDNb19hyHQISEYlTKgARkTilAhARiVMqABGROKUCEBGJUxEVgJlNMLPlZrbSzG5q5PYkM3squH2OmWUFy8ea2fzgZ4GZXRQs72dmb5vZUjPLN7PvNeeTEhGRAztgAZhZAnA/cC6QDVxmZtkNhl0N7HT3wcC9wN3B8sVAjruPASYAfzWzRKAa+KG7DwdOAL7dyGOKiEgLimQPYCyw0t1Xu3sl8CQwscGYicCM4PIzwJlmZu5e5u7VwfJkwAHcfZO7fxRcLgWWAn0P76mISFi27Crnz/9cxQvzNzC/oJjissqwI0kEIvkgWF+goN71QmBcU2PcvdrMSoB0YLuZjQOmAwOAK+oVAgDB4aJjgDmN/XIzmwpMBejfv38EcUWktd383CJmL9v6iWXdkhPJ6t6ZAemdyUrvtP/P/umdyOiShJmFlFb2iaQAGvtbavgtMk2Ocfc5wAgzGw7MMLNX3b0cwMy6AM8C33f3XY39cnefBkwDyMnJ0bfXiESZt5ZuYfayrdw4YRhnD+/J2h1lrNuxh3U7yli7Yw8LCop5eeFGauv97+3UIeFTxTAgvTMD0jvRq1sy7dqpHFpDJAVQCPSrdz0T2NjEmMLgGH8KUFR/gLsvNbM9wEgg18zaU7fxf9zdnzvE/CISovKqGu58aQmDe3Thm58bRPuEdgzp2fVT4yqra9lQvJe1O/awbvse1hWVsW5HGcu3lPLm0i1U1fy3HToktmNAWr1i6N6ZAWmdyErvTJ/UZBITNHmxuURSAHOBIWY2ENgATAK+2mDMLGAK8D5wCTDb3T24T0FwWGgAMAxYa3X7fg8BS939t830XESklT347mrW7SjjsavH0f4zNswdEtsxsHtnBnbvXLcVqKem1tlYvJd1O8pYVxTsOWyv+/PfK7dRXlW7f2xiO6NfWicGpNcVQv+0TmR1ryuLfkd0okOiyuFgHLAAgo339cDrQAIw3d3zzexOINfdZ1G3MZ9pZiupe+U/Kbj7KcBNZlYF1ALXuft2MzsFuAJYZGbzg7G3uPsrzfrsRKTFbCjeyx/fXsm5I3txypDuh/w4CcFGvV9aJ07hk49TW+tsLa1g7Y49rA8OKe07tJS7die7K/77lmI7gz6pHeuKIb1TvcNLdUXRsUPCIWdsqyyWvhQ+JyfHdTZQkejw7cc/4q1lW3jzhlPJPKJTq/9+d2fHnsp67zfUvfew78/isqpPjO/VLfkTxTC8d1dGZ6aS3iWp1bO3NjPLc/echstj6nTQIhId/rNyOy8v2sQNZw8NZeMPYGZ075JE9y5JHDcg7VO3l5RVsa4oKITt/y2G2cu2sX134f5x/dI6MjozlTH9UhndL5WRfVLiZm9BBSAiB6WqppbbZuXTP60TUz8/KOw4TUrp1J5RnVIZlZn6qdtKy6tYsnEX8wuKWVBYzLz1xby0cBNQd0hqWM+ujO6Xyph+KYzul8qQHl1JaIMzk1QAInJQHn1/HSu27uaByTkkt4/NV8pdk9szblA64wal71+2tbSchQUlLCgsZn4wdfWJD9cDddNWj+6bsn8vYXS/VPqkJMf8ZxlUACISsW2lFfzujY85dWgGZw3vEXacZtWjazJnZSdzVnZPoO4N6LU79rCgsJgFBSXMLyjm4ffWUllTNyupe5ckxtTbSxjVN5WUTu3DfAoHTQUgIhG7+7VllFfXcNsF2TH/6vdA2rUzBmV0YVBGFy46JhOAiuoalm0q3b+XsKCgmDeXbtl/n0HdOweHjur2Eob37kpSYvTuJakARCQieet28kxeIdeeeiSDMrqEHScUSYkJ+w8BTT6xblnJ3ioWb6jbQ5hfUMy/V27n+XkbAGifYGT37lZ3n8xUxvRPZWB656j5pLOmgYrIAdXUOl+8/z22lpYz+4en0TlJrx2b4u5s3lXOgoJi5heUML9gJ4sKS9hTWQNA1+RERmemMrpfyv7ZRz26JbdoJk0DFZFD9nRuAYs2lHDfpDHa+B+AmdE7pSO9UzoyYWRvoK5AV23bvf+w0YLCYv76r9VUBydI6pOSvH/PYnRmKkdnptClFdaz/iZF5DMVl1Vyz2vLGDswjQtH9wk7TkxKaGcM7dmVoT278pWculOrlVfVkL+xhPkFJftL4dXFmwEwgyE9uvx31lFmKsN7d2v2qagqABH5TL9942NK9lZxx4Uj2vwbv60puX0Cxw1I+8SH2HbuqWzwBvNWns6t+9DagtvGk9KxeWcZqQBEpElLNu7isQ/WMfnELIb37hZ2nDbviM4dOG1YD04bVjfF1t0p3LmX5ZtLm33jDyoAEWmCu3PbrMWkdurAD84aGnacuGT23xPltQSdO1VEGvXC/I3MXbuTn0wYFnMfcJLIqABE5FN2V1Rz1ytLGZ2ZwpeP63fgO0hM0iEgEfmUP7y1gq2lFUybnBM1H1qS5qc9ABH5hJVbdzP9vTV8JSeTMf0+fSZNaTtUACKyn7tzx4v5JLdP4MYJR4UdR1qYCkBE9vvHki28u2I7N5w9lO5x8E1Z8U4FICJA3SdT//elJQzr2ZUrThgQdhxpBXoTWEQA+Mu/VlG4cy9PTj2BxAS9NowH+lsWEQqKyvjzP1dxweg+nFDvW7KkbVMBiAg/f3kJ7cy45Qt64zeeqABE4tw7H2/j9fwtfOfMwfRO6Rh2HGlFKgCROFZZXcvtL+YzsHtnrj5lYNhxpJWpAETi2MPvrWH1tj387PzsqP7uWmkZKgCROLVlVzm/f2sFZw3vwelH9Qg7joRABSASp375ylKqap1bz88OO4qERAUgEoc+XFPE3+dv5JrPD2JAeuew40hIIioAM5tgZsvNbKWZ3dTI7Ulm9lRw+xwzywqWjzWz+cHPAjO7qN59ppvZVjNb3FxPRkQOrKbWuW1WPn1SkrnutMFhx5EQHbAAzCwBuB84F8gGLjOzhvuMVwM73X0wcC9wd7B8MZDj7mOACcBfzWzfp48fCZaJSCv6f3PWsXTTLv7n/Gw6dtAbv/Eskj2AscBKd1/t7pXAk8DEBmMmAjOCy88AZ5qZuXuZu1cHy5MB33cHd38HKDqs9CJyUIr2VPKbf3zMSUemc+7IXmHHkZBFUgB9gYJ61wuDZY2OCTb4JUA6gJmNM7N8YBFwbb1CiIiZTTWzXDPL3bZt28HcVUQa+PXry9lTUc0dF47ATF/0Eu8iKYDG/pV4pGPcfY67jwCOB242s+SDCeju09w9x91zMjIyDuauIlLPwsJinpy7niknZTGkZ9ew40gUiKQACoH6XwqaCWxsakxwjD+FBod33H0psAcYeahhReTQ1AZv/KZ3TuJ7Zw0JO45EiUgKYC4wxMwGmlkHYBIwq8GYWcCU4PIlwGx39+A+iQBmNgAYBqxtluQiErFnPypk3vpibjr3KLoltw87jkSJAxZAcMz+euB1YCnwtLvnm9mdZnZhMOwhIN3MVgI3APumip4CLDCz+cDzwHXuvh3AzJ4A3geGmVmhmV3dnE9MROrsKq/i7teWcWz/VL50TMO37ySeRfSFMO7+CvBKg2U/q3e5HPhyI/ebCcxs4jEvO6ikInJIfvfGCnbsqeSRq8bSrp3e+JX/0ieBRdqwj7eUMuP9tVw2tj8j+6aEHUeijApApI1yd26flU+XpER+PH5Y2HEkCqkARNqoVxZt5j+rdvCjc4ZxROcOYceRKKQCEGmDyiqr+cXLS8ju3Y2vju0fdhyJUioAkTboT2+vYmNJOXdOHEGC3viVJqgARNqYtdv3MO2d1Vx0TF9ystLCjiNRTAUg0sb870tLaJ9g3HzuUWFHkSinAhBpQ2Yv28Jby7byvbOG0KPbQZ12S+KQCkCkjaioruHOF5cwKKMzV540MOw4EgMi+iSwiES/B99dw9odZTz69bF0SNRrOzkw/SsRaQM2Fu/lj7NXcs6Innx+qE6bLpFRAYi0AXe9spRad/7nvIbf1irSNBWASIz7z6rtvLRwE9867Uj6pXUKO47EEBWASAyrqqnljllLyDyiI9eeemTYcSTGqABEYtjM99exfEspt56fTXL7hLDjSIxRAYjEqG2lFdz7xsd8fmgG47N7hh1HYpAKQCRG3fPaMsqra7jtgmzMdL4fOXgqAJEYNG/9Tv6WV8jXTxnIkRldwo4jMUoFIBJjamud22bl06NrEt85Y0jYcSSGqQBEYszTuQUsLCzhp+cNp0uSPswvh04FIBJDSsqquOf15YzNSuPC0X3CjiMxTgUgEkN++8Zyissquf3CEXrjVw6bCkAkRizZuIuZH6zj8hMGkN2nW9hxpA1QAYjEAHfn9ln5pHbqwA1nDw07jrQRKgCRGDBrwUY+XFvEj88ZRmqnDmHHkTZCBSAS5XZXVHPXK0sZlZnCV3L6hR1H2hDNIROJcn+YvYItuyr4y+XHkdBOb/xK89EegEgUW7VtN9P/vYYvH5fJMf2PCDuOtDERFYCZTTCz5Wa20sxuauT2JDN7Krh9jpllBcvHmtn84GeBmV0U6WOKxDt3544Xl5CcmMCNE44KO460QQcsADNLAO4HzgWygcvMrOHXDl0N7HT3wcC9wN3B8sVAjruPASYAfzWzxAgfUySuvbFkC+98vI3vnz2UjK5JYceRNiiSPYCxwEp3X+3ulcCTwMQGYyYCM4LLzwBnmpm5e5m7VwfLkwE/iMcUiVuv52/mlucXMbRnFyafOCDsONJGRVIAfYGCetcLg2WNjgk2+CVAOoCZjTOzfGARcG1weySPSXD/qWaWa2a527ZtiyCuSOzasquca2fmcc3MPLp3SeKPXz2W9gl6q05aRiSzgBqbduCRjnH3OcAIMxsOzDCzVyN8TIL7TwOmAeTk5DQ6RiTW1dY6T8xdz69eXUZldS03ThjGNz83SBt/aVGRFEAhUH/ycSawsYkxhWaWCKQARfUHuPtSM9sDjIzwMUXiwsqtu7nluUV8uLaIk45M566Ljiare+ewY0kciKQA5gJDzGwgsAGYBHy1wZhZwBTgfeASYLa7e3CfAnevNrMBwDBgLVAcwWOKtGmV1bX8+Z+ruP/tlXTskMA9l4ziy8dl6iRv0moOWADBxvt64HUgAZju7vlmdieQ6+6zgIeAmWa2krpX/pOCu58C3GRmVUAtcJ27bwdo7DGb+bmJRK28dUXc9OwiVmzdzQWj+/Cz87M100danbnHzmH1nJwcz83NDTuGyCErLa/inteW89icdfRJ6cjPvziS04/qEXYsaePMLM/dcxou16kgRFrJP/I387MX8tlSWs6VJ2Xxo/HD6Kxv9JIQ6V+fSAvbuquc22bl8+rizRzVqyt/vvxYndZBooIKQKSF1NY6T84t4JevLqWiupYfnzOMqZ/X1E6JHioAkRawattubn5uER+uKeKEQWncddHRDMroEnYskU9QAYg0o8rqWv76r1X8YfZKktu34+6Lj+YrOf00tVOikgpApJl8tH4nNz+7iOVbSjlvVG9uuyCbHl2Tw44l0iQVgMhh2l1Rza9fW8ajH6yjV7dkHpqSw5nDe4YdS+SAVAAih+HNJVu49YXFbN5VzpQTs/jROcPooqmdEiP0L1XkEGwtLeeOWUt4edEmhvXsyv1fO5ZjNbVTYowKQOQguDtPzS3grleWUl5dy4/GD2Xq54+kQ6KmdkrsUQGIRGh1MLVzzpoixg1M45df0tROiW0qAJEDqKyuZdo7q/j97JUkJbbjV1+qm9rZrp2mdkpsUwGIfIZ563dyUzC18wtH9+L2C0bQo5umdkrboAIQacTuimp+8/pyZry/lp5dk3lgcg5nZ2tqp7QtKgCRBt5auoVb/76YTbvKueKEAfz4nGF0TW4fdiyRZqcCEAlsK63gjhfzeWnhJob27MIzXz2J4wZoaqe0XSoAiXvuzt9yC/nFK0vZW1nDDWcP5dpTNbVT2j4VgMS1Ndv3cMtzi3h/9Q7GZqVx15eOZnAPTe2U+KACkLhUVVPLtHdW8/u3VtAhsR13XXQ0k47X1E6JLyoAiTvzC4q56dmFLNtcyrkje3HHhZraKfFJBSBxwd1Zu6OMR99fy4z/rKVH12SmXXEc40f0CjuaSGhUANImuTurtu3mg9VFzFlTxJzVO9haWoEZXD5uADdO0NROERWAtAm1tc7yLaXMWb2DOWuK+HBNETv2VALQs1sSJwxKZ9ygNE4+sjtZ3TuHnFYkOqgAJCbV1DpLN+3ig2CDP3dtEcVlVQD0Te3IqUMzGDcojXED0xmQ3klfySjSCBWAxISqmloWbyjZfzgnd+1OSiuqARiQ3onx2T0ZN7DuVX7mEZ1CTisSG1QAEpUqqmtYVFi3wf9g9Q7y1u2krLIGgEEZnTl/dB9OCF7h90rRDB6RQ6ECkKhQXlXDvPXFzFmzgzmri/ho/U4qqmsBGNazK5ccl8m4gekcP/AIfdG6SDNRAUgoyiqryVu3kw/XFDFndRHzC4qprKnFDIb36sZXx/Vn3MB0xg5MI61zh7DjirRJERWAmU0A7gMSgAfd/VcNbk8CHgWOA3YAl7r7WjM7G/gV0AGoBH7s7rOD+1wK/DR4zJfd/cbmeUoSjUrLq8hdt5M5q4uYs2YHiwpLqK51EtoZI/t048qTsxg3MI2crDRSOmp6pkhrOGABmFkCcD9wNlAIzDWzWe6+pN6wq4Gd7j7YzCYBdwOXAtuBC9x9o5mNBF4H+ppZOvBr4Dh332ZmM8zsTHd/q3mfnoSlpKyKD9cW8eGaulk6izeUUOuQ2M4YlZnCNz8/aP8Gv0uSdkRFwhDJ/7yxwEp3Xw1gZk8CE4H6BTARuD24/AzwRzMzd59Xb0w+kBzsLQwCPnb3bcFtbwIXAyqAGFW0p5IP1+zY/8GrZZt34Q4dEtoxpn8q158+mLED0zl2QCqdOmiDLxINIvmf2BcoqHe9EBjX1Bh3rzazEiCduj2AfS4G5rl7hZmtBI4ys6zg8b5I3WGiTzGzqcBUgP79+0cQV1pLdU0tf3x7Ja8s2sTHW3YDkNy+Hcf2P4LvnzmUcYPSGNMvleT2CSEnFZHGRFIAjX2Cxg9mjJmNoO6w0HgAd99pZt8CngJqgf9Qt1fw6QdxnwZMA8jJyWn4eyUkJWVVXP/ER7y7YjsnHZnOxDF9GTcwjVGZqTqPvkiMiKQACoF+9a5nAhubGFNoZolAClAEYGaZwPPAZHdfte8O7v4i8GIwZipQc4jPQVrZyq2lfGNGLhuK93LPxaP4yvH9DnwnEYk6kbxUmwsMMbOBZtYBmATMajBmFjAluHwJMNvd3cxSgZeBm939vfp3MLMewZ9HANcBDx7605DW8vayrVx0/3/YXVHNE988QRt/kRh2wD2A4Jj+9dTN4EkAprt7vpndCeS6+yzgIWBmcGy/iLqSALgeGAzcama3BsvGu/tW4D4zGx0su9PdP26+pyXNzd154N3V/PLVZQzv1Y0HpuTQN7Vj2LFE5DCYe+wcVs/JyfHc3NywY8Sd8qoabnluEc/N28AXju7Fb748WjN5RGKImeW5e07D5fpfLJ9p665yps7MY35BMT84ayjfPXOwzqwp0kaoAKRJCwuLmfpoHiV7q/jL5ccyYWTvsCOJSDNSAUijXpi/gRufWUj3Lkk8+62TyO7TLexIItLMVADyCbW1zv+9sZz7317F2Kw0/nT5sXTvkhR2LBFpASoA2W93RTXff3I+by7dwqTj+3HnxJH6UJdIG6YCEADW7yjjG4/OZdW2Pdx+QTZTTsrSm70ibZwKQHh/1Q6uezyPWocZV43llCHdw44kIq1ABRDnZn6wjjtm5ZPVvTMPTs4hq3vnsCOJSCtRAcSpqppabp+Vz+Nz1nP6sAzuu+wYuiXri1hE4okKIA4V7ankusfz+GB1EdecOogbzzmKhHY63i8Sb1QAcWb55lK+8ehctuyq4N5LR3PRMZlhRxKRkKgA4sg/8jfzg6fm0zkpkaevOZEx/VLDjiQiIVIBxAF350//XMVv/rGco/umMO2KHHqlJIcdS0RCpgJo4/ZW1nDjswt5ccFGJo7pw90Xj9JXNIoIoAJo0zaV7GXqo3ks3ljCTyYcxbWnDtKHu0RkPxVAG/XR+p1cMzOPsopqHrgih7Oye4YdSUSijAqgDXo2r5Cbn1tEr5RkHv/GOIb27Bp2JBGJQiqANqSm1rn7tWVMe2c1Jw5K509fO5YjOncIO5aIRCkVQBuxq7yK7z4xj38u38bkEwdw6/nZtE/QmTxFpGkqgDZgzfY9XD1jLut3lPGLi0bytXEDwo4kIjFABRDj3l2xjW8//hEJ7YzHvjGOEwalhx1JRGKECiBGuTuP/GctP395KUN6dOGByTn0S+sUdiwRiSEqgBhUUV3Dz/6ez1O5BYzP7sm9l46hc5L+KkXk4GirEWO2767g2pl55K7byXfOGMwPzhpKO53JU0QOgQoghuRvLGHqo3ns2FPBHy47hgtG9wk7kojEMBVAjHh10SZueHoBqZ3a88y1JzGyb0rYkUQkxqkAolxtrXPfWyu4760VHNs/lb9ccRw9uupMniJy+FQAUaysspofPr2AVxdv5pLjMvnFRSNJStSZPEWkeUT0UVEzm2Bmy81spZnd1MjtSWb2VHD7HDPLCpafbWZ5ZrYo+POMeve5LFi+0MxeM7PuzfWk2oLCnWVc/Of3eT1/M/9z3nB+fckobfxFpFkdsADMLAG4HzgXyAYuM7PsBsOuBna6+2DgXuDuYPl24AJ3PxqYAswMHjMRuA843d1HAQuB6w//6bQNc9cWMfGP71G4s4zpVx7PNz6n0ziLSPOLZA9gLLDS3Ve7eyXwJDCxwZiJwIzg8jPAmWZm7j7P3TcGy/OBZDNLAiz46Wx1W7ZuwEaEN5ds4asPfEBKx/b8/dsnc9qwHmFHEpE2KpIC6AsU1LteGCxrdIy7VwMlQMNzElwMzHP3CnevAr4FLKJuw58NPNTYLzezqWaWa2a527ZtiyBu7FpQUMz1T3xEdu9uPH/dyRyZ0SXsSCLShkVSAI0de/CDGWNmI6g7LHRNcL09dQVwDNCHukNANzf2y919mrvnuHtORkZGBHFjU0FRGVfPyCWjaxIPTjmelE7tw44kIm1cJAVQCPSrdz2TTx+u2T8mOL6fAhQF1zOB54HJ7r4qGD8GwN1XubsDTwMnHeJziHklZVVc9chcKqtrePjKsWR0TQo7kojEgUgKYC4wxMwGmlkHYBIwq8GYWdS9yQtwCTDb3d3MUoGXgZvd/b164zcA2Wa27yX92cDSQ30SsayyupZrHstl/Y4ypk3OYXAPHfYRkdZxwM8BuHu1mV0PvA4kANPdPd/M7gRy3X0WdcfvZ5rZSupe+U8K7n49MBi41cxuDZaNd/eNZnYH8I6ZVQHrgCub84nFAnfnpmcX8sHqIu6bNEanchaRVmV1R2BiQ05Ojufm5oYdo9n89o2P+f1bK/jR+KFcf8aQsOOISBtlZnnuntNwub4zMCR/yy3g92+t4NKcfnz79MFhxxGROKQCCMG/V2zn5ucW8bkh3fn5RSP1IS8RCYUKoJUt31zKtx7LY3CPLtz/tWP1xe0iEhptfVrRll3lXPXwh3RKSmD6lcfTLVlz/UUkPCqAVrKnopqrZ8ylZG8V0688nj6pHcOOJCJxTqeDbgXVNbV854l5LN1UyoNTchjRR1/mIiLh0x5AC3N3bn8xn9nLtnLnxBGcrpO7iUiUUAG0sAffXcNjH6zn2lOP5GvjBoQdR0RkPxVAC3pl0SZ+8cpSzhvVmxvPGRZ2HBGRT1ABtJC8dUV8/6n55Aw4gv/78mjatdNcfxGJLiqAFrB2+x6++WgefVM7Mm1yDsnt9VWOIhJ9VADNrGhPJVc9Mhd35+Erjyetc4ewI4mINErTQJtReVUNUx/NZUPxXp745jiyuncOO5KISJO0B9BMamudH/1tAbnrdvK7S8dw3IC0sCOJiHwmFUAzuef15by0cBO3fOEovnB077DjiIgckAqgGTw+Zx1/+dcqLj+hP9/83KCw44iIREQFcJjeXr6Vn72QzxlH9eD2C0bo1M4iEjNUAIchf2MJ1z/+EUf16sofLjuGRJ3aWURiiLZYh2hj8V6+/shcUjq2Z/qVx9M5SROqRCS2aKt1CErLq/j6I3Mpq6jhmW+dRM9uyWFHEhE5aCqAg1RVU8t1j3/Eyq27eeSqsQzr1TXsSCIih0QFcBDcnf95fjHvrtjOPZeM4pQh3cOOJCJyyPQewEH40z9X8VRuAd89YzBfyekXdhwRkcOiAojQC/M38OvXl/OlY/ryg7OHhh1HROSwqQAi8MHqHfz4bws5YVAav7p4lOb6i0iboAI4gJVbd3PNzDz6p3fir5fn0CFRq0xE2gZtzT7DttIKrnrkQ9onGA9feTwpndqHHUlEpNloFlAT9lbW8I1Hc9lWWsFTU0+kX1qnsCOJiDSriPYAzGyCmS03s5VmdlMjtyeZ2VPB7XPMLCtYfraZ5ZnZouDPM4LlXc1sfr2f7Wb2u+Z8Yoejptb5/lPzWFhYzO8nHcPofqlhRxIRaXYH3AMwswTgfuBsoBCYa2az3H1JvWFXAzvdfbCZTQLuBi4FtgMXuPtGMxsJvA70dfdSYEy935EHPNdcT+pw/eLlpbyev4XbL8hm/IheYccREWkRkewBjAVWuvtqd68EngQmNhgzEZgRXH4GONPMzN3nufvGYHk+kGxmSfXvaGZDgB7Au4f6JJrTw++tYfp7a/j6yahgNL0AAAajSURBVAO58uSBYccREWkxkRRAX6Cg3vXCYFmjY9y9GigB0huMuRiY5+4VDZZfBjzl7h5p6Jbyj/zN3PnSEsZn9+Sn5w0PO46ISIuK5E3gxia9N9xYf+YYMxtB3WGh8Y2MmwRc0eQvN5sKTAXo37//gbIesgUFxXz3yXmMykzlvknHkNBOc/1FpG2LZA+gEKh/3oNMYGNTY8wsEUgBioLrmcDzwGR3X1X/TmY2Gkh097ymfrm7T3P3HHfPycjIiCDuwSsoKuPqGXPJ6JrEg5Nz6NghoUV+j4hINImkAOYCQ8xsoJl1oO4V+6wGY2YBU4LLlwCz3d3NLBV4GbjZ3d9r5LEvA544tOjNo6SsiqsemUtldS0PXzmWjK5JB76TiEgbcMACCI7pX0/dDJ6lwNPunm9md5rZhcGwh4B0M1sJ3ADsmyp6PTAYuLXelM8e9R7+K4RYABXVNVzzWC7rd5QxbXIOg3t0CSuKiEirsyh47zViOTk5npub2yyP5e7c8PQCnp+3gfsmjWHimIbva4uItA1mlufuOQ2Xx+2pIO59cwXPz9vAj8YP1cZfROJSXBbA07kF/P6tFVya049vnz447DgiIqGIuwL494rt3PLcIj43pDs/v2ikTu0sInErrgpg+eZSvvVYHoN7dOH+rx1L+4S4evoiIp8QN1vALbvKuerhD+mUlMD0K4+nW7JO7Swi8S0uCmBPRTVff2QuJXurmH7l8fRJ7Rh2JBGR0LX57wOorqnlO0/MY9nmUh6cksOIPilhRxIRiQptvgDamXFkRmfOHN6D04f1OPAdRETiRNsvgHbGT8/LDjuGiEjUiYv3AERE5NNUACIicUoFICISp1QAIiJxSgUgIhKnVAAiInFKBSAiEqdUACIicSqmvhHMzLYB6w7x7t2B7c0YpyXFUlaIrbyxlBViK28sZYXYynu4WQe4e0bDhTFVAIfDzHIb+0q0aBRLWSG28sZSVoitvLGUFWIrb0tl1SEgEZE4pQIQEYlT8VQA08IOcBBiKSvEVt5YygqxlTeWskJs5W2RrHHzHoCIiHxSPO0BiIhIPSoAEZE41SYLwMymm9lWM1tcb1mamb1hZiuCP48IM+M+TWS93cw2mNn84OcLYWbcx8z6mdnbZrbUzPLN7HvB8mhdt03ljbr1a2bJZvahmS0Ist4RLB9oZnOCdfuUmXUIOyt8Zt5HzGxNvXU7Juys+5hZgpnNM7OXgutRuW6h0awtsl7bZAEAjwATGiy7CXjL3YcAbwXXo8EjfDorwL3uPib4eaWVMzWlGvihuw8HTgC+bWbZRO+6bSovRN/6rQDOcPfRwBhggpmdANxNXdYhwE7g6hAz1tdUXoAf11u388OL+CnfA5bWux6t6xY+nRVaYL22yQJw93eAogaLJwIzgsszgC+2aqgmNJE1Krn7Jnf/KLhcSt0/0L5E77ptKm/U8Tq7g6vtgx8HzgCeCZZH07ptKm9UMrNM4DzgweC6EaXrtmHWltQmC6AJPd19E9RtGIBo/4b4681sYXCIKCoOqdRnZlnAMcAcYmDdNsgLUbh+g93++cBW4A1gFVDs7tXBkEKiqMAa5nX3fev2F8G6vdfMkkKMWN/vgBuB2uB6OtG7bhtm3afZ12s8FUAs+TNwJHW71puA/ws3zieZWRfgWeD77r4r7DwH0kjeqFy/7l7j7mOATGAsMLyxYa2bqmkN85rZSOBm4CjgeCAN+EmIEQEws/OBre6eV39xI0NDX7dNZIUWWq/xVABbzKw3QPDn1pDzNMndtwT/uWqBB6jbGEQFM2tP3cb0cXd/Llgcteu2sbzRvH4B3L0Y+Cd171ukmllicFMmsDGsXE2pl3dCcNjN3b0CeJjoWLcnAxea2VrgSeoO/fyO6Fy3n8pqZo+11HqNpwKYBUwJLk8BXggxy2fatzENXAQsbmpsawqOmz4ELHX339a7KSrXbVN5o3H9mlmGmaUGlzsCZ1H3nsXbwCXBsGhat43lXVbvhYBRd0w99HXr7je7e6a7ZwGTgNnu/jWicN02kfXyllqviQceEnvM7AngNKC7mRUCtwG/Ap42s6uB9cCXw0v4X01kPS2Y5uXAWuCa0AJ+0snAFcCi4NgvwC1E6bql6byXReH67Q3MMLME6l6YPe3uL5nZEuBJM/s5MI+6QosGTeWdbWYZ1B1imQ9cG2bIA/gJ0bluG/N4S6xXnQpCRCROxdMhIBERqUcFICISp1QAIiJxSgUgIhKnVAAiInFKBSAiEqdUACIicer/AxONdXw6bSIAAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "\n",
    "plt.plot(K,SSE)\n",
    "plt.savefig('silhouette.png')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "kMeans done, loss = 3.22\n",
      "2019-10-28 12:33:35.551351\n"
     ]
    }
   ],
   "source": [
    "# 20 clusters\n",
    "%%time\n",
    "n_cluster = 20\n",
    "km_cluster = KMeans(n_clusters=n_cluster).fit(word_vecs)\n",
    "d = distance.cdist(word_vecs,km_cluster.cluster_centers_,metric='euclidean')\n",
    "SS = np.sum(np.min(d,axis=1))/len(word_vecs)\n",
    "print('kMeans done, loss = {:.3}'.format(SS))\n",
    "print(datetime.now())\n",
    "\n",
    "km_labels = km_cluster.labels_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.DataFrame(word_vecs,columns = [i for i in range(300)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.index = vocab"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "df['label'] = km_labels"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv('word_class20.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "# 35 clusters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "kMeans done, loss = 3.15\n"
     ]
    },
    {
     "ename": "NameError",
     "evalue": "name 'df' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<timed exec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'df' is not defined"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "n_cluster = 35\n",
    "km_cluster = KMeans(n_clusters=n_cluster).fit(word_vecs)\n",
    "d = distance.cdist(word_vecs,km_cluster.cluster_centers_,metric='euclidean')\n",
    "SS = np.sum(np.min(d,axis=1))/len(word_vecs)\n",
    "print('kMeans done, loss = {:.3}'.format(SS))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>291</th>\n",
       "      <th>292</th>\n",
       "      <th>293</th>\n",
       "      <th>294</th>\n",
       "      <th>295</th>\n",
       "      <th>296</th>\n",
       "      <th>297</th>\n",
       "      <th>298</th>\n",
       "      <th>299</th>\n",
       "      <th>label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>what</td>\n",
       "      <td>0.112631</td>\n",
       "      <td>-0.027567</td>\n",
       "      <td>0.389141</td>\n",
       "      <td>-0.106166</td>\n",
       "      <td>-0.036858</td>\n",
       "      <td>0.174021</td>\n",
       "      <td>0.183803</td>\n",
       "      <td>0.082252</td>\n",
       "      <td>0.007311</td>\n",
       "      <td>-0.336434</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.044195</td>\n",
       "      <td>-0.256803</td>\n",
       "      <td>-0.103212</td>\n",
       "      <td>0.168220</td>\n",
       "      <td>-0.013151</td>\n",
       "      <td>-0.305558</td>\n",
       "      <td>-0.002310</td>\n",
       "      <td>0.084851</td>\n",
       "      <td>-0.058029</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>we</td>\n",
       "      <td>0.256392</td>\n",
       "      <td>0.059434</td>\n",
       "      <td>0.071496</td>\n",
       "      <td>0.042644</td>\n",
       "      <td>-0.044782</td>\n",
       "      <td>-0.123316</td>\n",
       "      <td>0.084929</td>\n",
       "      <td>-0.120551</td>\n",
       "      <td>-0.134421</td>\n",
       "      <td>-0.061319</td>\n",
       "      <td>...</td>\n",
       "      <td>0.024443</td>\n",
       "      <td>-0.089100</td>\n",
       "      <td>-0.016137</td>\n",
       "      <td>0.119141</td>\n",
       "      <td>0.007926</td>\n",
       "      <td>-0.089660</td>\n",
       "      <td>-0.041727</td>\n",
       "      <td>-0.091241</td>\n",
       "      <td>-0.003752</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>know</td>\n",
       "      <td>0.155820</td>\n",
       "      <td>0.021244</td>\n",
       "      <td>0.045042</td>\n",
       "      <td>-0.004687</td>\n",
       "      <td>-0.103851</td>\n",
       "      <td>-0.074655</td>\n",
       "      <td>0.114026</td>\n",
       "      <td>-0.137674</td>\n",
       "      <td>-0.139760</td>\n",
       "      <td>-0.006807</td>\n",
       "      <td>...</td>\n",
       "      <td>0.050255</td>\n",
       "      <td>-0.182831</td>\n",
       "      <td>-0.040470</td>\n",
       "      <td>0.144936</td>\n",
       "      <td>-0.015766</td>\n",
       "      <td>-0.177883</td>\n",
       "      <td>-0.042206</td>\n",
       "      <td>-0.042344</td>\n",
       "      <td>0.024171</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>and</td>\n",
       "      <td>0.169945</td>\n",
       "      <td>0.009516</td>\n",
       "      <td>0.049934</td>\n",
       "      <td>0.196669</td>\n",
       "      <td>-0.140271</td>\n",
       "      <td>-0.025416</td>\n",
       "      <td>0.145185</td>\n",
       "      <td>-0.058401</td>\n",
       "      <td>-0.144195</td>\n",
       "      <td>-0.139730</td>\n",
       "      <td>...</td>\n",
       "      <td>0.030937</td>\n",
       "      <td>-0.046990</td>\n",
       "      <td>0.105125</td>\n",
       "      <td>0.130561</td>\n",
       "      <td>0.137056</td>\n",
       "      <td>0.097826</td>\n",
       "      <td>0.131815</td>\n",
       "      <td>0.038892</td>\n",
       "      <td>0.038345</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>do_not</td>\n",
       "      <td>0.180850</td>\n",
       "      <td>-0.057114</td>\n",
       "      <td>0.159603</td>\n",
       "      <td>0.092293</td>\n",
       "      <td>-0.113717</td>\n",
       "      <td>-0.077967</td>\n",
       "      <td>0.072090</td>\n",
       "      <td>-0.058054</td>\n",
       "      <td>-0.207193</td>\n",
       "      <td>-0.110683</td>\n",
       "      <td>...</td>\n",
       "      <td>0.119232</td>\n",
       "      <td>-0.166349</td>\n",
       "      <td>0.018261</td>\n",
       "      <td>0.066868</td>\n",
       "      <td>0.117246</td>\n",
       "      <td>-0.171014</td>\n",
       "      <td>0.139693</td>\n",
       "      <td>0.050467</td>\n",
       "      <td>-0.002439</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 301 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               0         1         2         3         4         5         6  \\\n",
       "what    0.112631 -0.027567  0.389141 -0.106166 -0.036858  0.174021  0.183803   \n",
       "we      0.256392  0.059434  0.071496  0.042644 -0.044782 -0.123316  0.084929   \n",
       "know    0.155820  0.021244  0.045042 -0.004687 -0.103851 -0.074655  0.114026   \n",
       "and     0.169945  0.009516  0.049934  0.196669 -0.140271 -0.025416  0.145185   \n",
       "do_not  0.180850 -0.057114  0.159603  0.092293 -0.113717 -0.077967  0.072090   \n",
       "\n",
       "               7         8         9  ...       291       292       293  \\\n",
       "what    0.082252  0.007311 -0.336434  ... -0.044195 -0.256803 -0.103212   \n",
       "we     -0.120551 -0.134421 -0.061319  ...  0.024443 -0.089100 -0.016137   \n",
       "know   -0.137674 -0.139760 -0.006807  ...  0.050255 -0.182831 -0.040470   \n",
       "and    -0.058401 -0.144195 -0.139730  ...  0.030937 -0.046990  0.105125   \n",
       "do_not -0.058054 -0.207193 -0.110683  ...  0.119232 -0.166349  0.018261   \n",
       "\n",
       "             294       295       296       297       298       299  label  \n",
       "what    0.168220 -0.013151 -0.305558 -0.002310  0.084851 -0.058029      4  \n",
       "we      0.119141  0.007926 -0.089660 -0.041727 -0.091241 -0.003752     32  \n",
       "know    0.144936 -0.015766 -0.177883 -0.042206 -0.042344  0.024171     32  \n",
       "and     0.130561  0.137056  0.097826  0.131815  0.038892  0.038345     32  \n",
       "do_not  0.066868  0.117246 -0.171014  0.139693  0.050467 -0.002439     32  \n",
       "\n",
       "[5 rows x 301 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "df35 = pd.DataFrame(word_vecs,columns = [i for i in range(300)])\n",
    "df35.index = vocab\n",
    "df35['label'] = km_cluster.labels_\n",
    "df35['label'].to_csv('word_class35.csv')\n",
    "df35.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### playground for 20 clusters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "df = pd.read_csv('word_class20.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 :  7856\n",
      "1 :  14167\n",
      "2 :  11109\n",
      "3 :  10735\n",
      "4 :  16233\n",
      "5 :  14158\n",
      "6 :  13030\n",
      "7 :  5435\n",
      "8 :  8505\n",
      "9 :  16897\n",
      "10 :  10966\n",
      "11 :  8707\n",
      "12 :  11801\n",
      "13 :  13394\n",
      "14 :  9625\n",
      "15 :  15752\n",
      "16 :  14649\n",
      "17 :  11683\n",
      "18 :  18009\n",
      "19 :  14790\n"
     ]
    }
   ],
   "source": [
    "for c in range(20):\n",
    "    print(c, ': ',len(df.loc[df['class'] == c]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "((247501, 2),          what  9\n",
       " 0          we  4\n",
       " 1        know  4\n",
       " 2         and  4\n",
       " 3      do_not  4\n",
       " 4  know_about  9)"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.shape, df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.columns = ['word','class']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>word</th>\n",
       "      <th>class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>150</td>\n",
       "      <td>bile_acid</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>158</td>\n",
       "      <td>bile_flow</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>171</td>\n",
       "      <td>bile_acid_bas</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>174</td>\n",
       "      <td>bas</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>219</td>\n",
       "      <td>gpr39</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>246</td>\n",
       "      <td>39_gpr39</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>442</td>\n",
       "      <td>gpr109a</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>451</td>\n",
       "      <td>beta_hydroxybutyrate</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>456</td>\n",
       "      <td>ketone_body</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>458</td>\n",
       "      <td>an_important_energy_source</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>516</td>\n",
       "      <td>gut_microbiota</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>518</td>\n",
       "      <td>gpr43</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>530</td>\n",
       "      <td>intestinal_microbiota</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>536</td>\n",
       "      <td>short_chain_fatty_acid_scfas</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>538</td>\n",
       "      <td>fermentation</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>539</td>\n",
       "      <td>dietary_fibre</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>549</td>\n",
       "      <td>scfas</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>552</td>\n",
       "      <td>fermentable</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>556</td>\n",
       "      <td>43_gpr43</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>557</td>\n",
       "      <td>ffar2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             word  class\n",
       "150                     bile_acid      0\n",
       "158                     bile_flow      0\n",
       "171                 bile_acid_bas      0\n",
       "174                           bas      0\n",
       "219                         gpr39      0\n",
       "246                      39_gpr39      0\n",
       "442                       gpr109a      0\n",
       "451          beta_hydroxybutyrate      0\n",
       "456                   ketone_body      0\n",
       "458    an_important_energy_source      0\n",
       "516                gut_microbiota      0\n",
       "518                         gpr43      0\n",
       "530         intestinal_microbiota      0\n",
       "536  short_chain_fatty_acid_scfas      0\n",
       "538                  fermentation      0\n",
       "539                 dietary_fibre      0\n",
       "549                         scfas      0\n",
       "552                   fermentable      0\n",
       "556                      43_gpr43      0\n",
       "557                         ffar2      0"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[df['class'] == 0].head(20) # inflammation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>word</th>\n",
       "      <th>class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>hematopoietic</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>248</td>\n",
       "      <td>hypermethylate</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>374</td>\n",
       "      <td>gpr56</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>376</td>\n",
       "      <td>proliferation</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>377</td>\n",
       "      <td>colorectal_cancer</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>379</td>\n",
       "      <td>metastasis</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>380</td>\n",
       "      <td>epithelialmesenchymal_transition</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>381</td>\n",
       "      <td>pi3k_akt_signal</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>382</td>\n",
       "      <td>g_proteincoupled</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>383</td>\n",
       "      <td>56_gpr56</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>387</td>\n",
       "      <td>oncogene</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>389</td>\n",
       "      <td>malignancy</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>393</td>\n",
       "      <td>colorectal_cancer_crc</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>401</td>\n",
       "      <td>quantitative_real_time</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>402</td>\n",
       "      <td>qrtpcr</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>405</td>\n",
       "      <td>prognostic_significance</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>407</td>\n",
       "      <td>survival</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>410</td>\n",
       "      <td>cell_count_kit8</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>418</td>\n",
       "      <td>transwell_assay</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>426</td>\n",
       "      <td>predict_poor_prognosis</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 word  class\n",
       "26                      hematopoietic      1\n",
       "248                    hypermethylate      1\n",
       "374                             gpr56      1\n",
       "376                     proliferation      1\n",
       "377                 colorectal_cancer      1\n",
       "379                        metastasis      1\n",
       "380  epithelialmesenchymal_transition      1\n",
       "381                   pi3k_akt_signal      1\n",
       "382                  g_proteincoupled      1\n",
       "383                          56_gpr56      1\n",
       "387                          oncogene      1\n",
       "389                        malignancy      1\n",
       "393             colorectal_cancer_crc      1\n",
       "401            quantitative_real_time      1\n",
       "402                            qrtpcr      1\n",
       "405           prognostic_significance      1\n",
       "407                          survival      1\n",
       "410                   cell_count_kit8      1\n",
       "418                   transwell_assay      1\n",
       "426            predict_poor_prognosis      1"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[df['class'] == 1].head(20) # cancer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>word</th>\n",
       "      <th>class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>81</td>\n",
       "      <td>allosteric</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>113</td>\n",
       "      <td>orthosteric_site</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>121</td>\n",
       "      <td>allosteric_ligand</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>123</td>\n",
       "      <td>topographically_distinct_site</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>128</td>\n",
       "      <td>orthosteric_ligand</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>307</td>\n",
       "      <td>key_step</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>371</td>\n",
       "      <td>slow_off_rate</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>588</td>\n",
       "      <td>pharmacophore_base_virtual_screening</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>590</td>\n",
       "      <td>virtual_screening</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>594</td>\n",
       "      <td>rational_drug_design</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>595</td>\n",
       "      <td>compound</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>597</td>\n",
       "      <td>pharmacophore</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>613</td>\n",
       "      <td>flexible</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>623</td>\n",
       "      <td>bind_mode</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>643</td>\n",
       "      <td>chemical_space</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>653</td>\n",
       "      <td>evaluation</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>680</td>\n",
       "      <td>chain_length</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1150</td>\n",
       "      <td>accommodate</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1201</td>\n",
       "      <td>functional_assay</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1228</td>\n",
       "      <td>weak_partial_agonist</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      word  class\n",
       "81                              allosteric      2\n",
       "113                       orthosteric_site      2\n",
       "121                      allosteric_ligand      2\n",
       "123          topographically_distinct_site      2\n",
       "128                     orthosteric_ligand      2\n",
       "307                               key_step      2\n",
       "371                          slow_off_rate      2\n",
       "588   pharmacophore_base_virtual_screening      2\n",
       "590                      virtual_screening      2\n",
       "594                   rational_drug_design      2\n",
       "595                               compound      2\n",
       "597                          pharmacophore      2\n",
       "613                               flexible      2\n",
       "623                              bind_mode      2\n",
       "643                         chemical_space      2\n",
       "653                             evaluation      2\n",
       "680                           chain_length      2\n",
       "1150                           accommodate      2\n",
       "1201                      functional_assay      2\n",
       "1228                  weak_partial_agonist      2"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[df['class'] == 2].head(20) # biased siangling"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>word</th>\n",
       "      <th>class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>83</td>\n",
       "      <td>7tm</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>92</td>\n",
       "      <td>seven_transmembrane_7tm_g_protein_couple</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>109</td>\n",
       "      <td>domain</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>189</td>\n",
       "      <td>gpcrs</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>263</td>\n",
       "      <td>dark</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>298</td>\n",
       "      <td>simulation</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>305</td>\n",
       "      <td>unbind</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>314</td>\n",
       "      <td>allosteric_communication_between</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>324</td>\n",
       "      <td>computational_method</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>325</td>\n",
       "      <td>metadynamic</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>326</td>\n",
       "      <td>markov_state_model</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>327</td>\n",
       "      <td>msms</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>331</td>\n",
       "      <td>motion</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>335</td>\n",
       "      <td>structural</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>346</td>\n",
       "      <td>helix</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>348</td>\n",
       "      <td>strand</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>354</td>\n",
       "      <td>conformational_change</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>355</td>\n",
       "      <td>disordering</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>365</td>\n",
       "      <td>mutant</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>370</td>\n",
       "      <td>unbinding</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         word  class\n",
       "83                                        7tm      3\n",
       "92   seven_transmembrane_7tm_g_protein_couple      3\n",
       "109                                    domain      3\n",
       "189                                     gpcrs      3\n",
       "263                                      dark      3\n",
       "298                                simulation      3\n",
       "305                                    unbind      3\n",
       "314          allosteric_communication_between      3\n",
       "324                      computational_method      3\n",
       "325                               metadynamic      3\n",
       "326                        markov_state_model      3\n",
       "327                                      msms      3\n",
       "331                                    motion      3\n",
       "335                                structural      3\n",
       "346                                     helix      3\n",
       "348                                    strand      3\n",
       "354                     conformational_change      3\n",
       "355                               disordering      3\n",
       "365                                    mutant      3\n",
       "370                                 unbinding      3"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[df['class'] == 3].head(20) # structure"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>word</th>\n",
       "      <th>class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>we</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>know</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>and</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>do_not</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>the</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>receptor</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>it</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>have</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>be</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>that</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>can</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>regulate</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>which</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>primarily</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>express</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>by</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>cell</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>of</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>system</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         word  class\n",
       "0          we      4\n",
       "1        know      4\n",
       "2         and      4\n",
       "3      do_not      4\n",
       "5         the      4\n",
       "7    receptor      4\n",
       "8           2      4\n",
       "10         it      4\n",
       "11       have      4\n",
       "12         be      4\n",
       "14       that      4\n",
       "16        can      4\n",
       "17   regulate      4\n",
       "20      which      4\n",
       "21  primarily      4\n",
       "22    express      4\n",
       "23         by      4\n",
       "24       cell      4\n",
       "25         of      4\n",
       "27     system      4"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[df['class'] == 4].head(20) # no-use"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>word</th>\n",
       "      <th>class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>162</td>\n",
       "      <td>intense</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>184</td>\n",
       "      <td>sensory_neuron</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>207</td>\n",
       "      <td>neuron</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>255</td>\n",
       "      <td>metabotropic</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>256</td>\n",
       "      <td>excitatory</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>257</td>\n",
       "      <td>inhibitory_neurotransmission</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>292</td>\n",
       "      <td>glutamate_release</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>352</td>\n",
       "      <td>network</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>403</td>\n",
       "      <td>immunohistochemistry</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>472</td>\n",
       "      <td>retina</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>481</td>\n",
       "      <td>situ_hybridization</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>622</td>\n",
       "      <td>input</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>695</td>\n",
       "      <td>taste</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>708</td>\n",
       "      <td>mouse_lack</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>741</td>\n",
       "      <td>somatosensory</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>754</td>\n",
       "      <td>give_rise</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>822</td>\n",
       "      <td>sensory_processing</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>842</td>\n",
       "      <td>innervate</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>887</td>\n",
       "      <td>confocal_laser_scan_microscopy</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>895</td>\n",
       "      <td>immunostaining</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               word  class\n",
       "162                         intense      5\n",
       "184                  sensory_neuron      5\n",
       "207                          neuron      5\n",
       "255                    metabotropic      5\n",
       "256                      excitatory      5\n",
       "257    inhibitory_neurotransmission      5\n",
       "292               glutamate_release      5\n",
       "352                         network      5\n",
       "403            immunohistochemistry      5\n",
       "472                          retina      5\n",
       "481              situ_hybridization      5\n",
       "622                           input      5\n",
       "695                           taste      5\n",
       "708                      mouse_lack      5\n",
       "741                   somatosensory      5\n",
       "754                       give_rise      5\n",
       "822              sensory_processing      5\n",
       "842                       innervate      5\n",
       "887  confocal_laser_scan_microscopy      5\n",
       "895                  immunostaining      5"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[df['class'] == 5].head(20) # neuroscience"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>word</th>\n",
       "      <th>class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>149</td>\n",
       "      <td>g_protein_couple</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>253</td>\n",
       "      <td>zn_2</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>294</td>\n",
       "      <td>together_our_finding_suggest_that</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>306</td>\n",
       "      <td>heterotrimeric_g_protein</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>309</td>\n",
       "      <td>signaling_cascade</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>339</td>\n",
       "      <td>rate_limit_step</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>344</td>\n",
       "      <td>translation</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>431</td>\n",
       "      <td>overexpression</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>558</td>\n",
       "      <td>here_we_show_that</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>657</td>\n",
       "      <td>decoy</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>675</td>\n",
       "      <td>heterodimer</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>709</td>\n",
       "      <td>wild_type</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>732</td>\n",
       "      <td>downstream</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>791</td>\n",
       "      <td>extracellular_signal_regulated_kinase_12</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>792</td>\n",
       "      <td>gq</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>793</td>\n",
       "      <td>g1213</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>805</td>\n",
       "      <td>plasma_membrane_localization</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>832</td>\n",
       "      <td>intracellular_signal_pathway</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>861</td>\n",
       "      <td>tyrosine_kinase</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>862</td>\n",
       "      <td>transactivation</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         word  class\n",
       "149                          g_protein_couple      6\n",
       "253                                      zn_2      6\n",
       "294         together_our_finding_suggest_that      6\n",
       "306                  heterotrimeric_g_protein      6\n",
       "309                         signaling_cascade      6\n",
       "339                           rate_limit_step      6\n",
       "344                               translation      6\n",
       "431                            overexpression      6\n",
       "558                         here_we_show_that      6\n",
       "657                                     decoy      6\n",
       "675                               heterodimer      6\n",
       "709                                 wild_type      6\n",
       "732                                downstream      6\n",
       "791  extracellular_signal_regulated_kinase_12      6\n",
       "792                                        gq      6\n",
       "793                                     g1213      6\n",
       "805              plasma_membrane_localization      6\n",
       "832              intracellular_signal_pathway      6\n",
       "861                           tyrosine_kinase      6\n",
       "862                           transactivation      6"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[df['class'] == 6].head(20) # signaling"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>word</th>\n",
       "      <th>class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>409</td>\n",
       "      <td>tumor</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>877</td>\n",
       "      <td>intensity_histogram</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>881</td>\n",
       "      <td>image</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1000</td>\n",
       "      <td>tumour</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1742</td>\n",
       "      <td>111</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2399</td>\n",
       "      <td>sst2</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2661</td>\n",
       "      <td>carcinoid_tumor</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2667</td>\n",
       "      <td>imaging</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2674</td>\n",
       "      <td>bronchopulmonary_carcinoid</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2679</td>\n",
       "      <td>125ityr10</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3715</td>\n",
       "      <td>high_resolution</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3763</td>\n",
       "      <td>misinterpretation</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4149</td>\n",
       "      <td>uptake</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4251</td>\n",
       "      <td>semiautomated</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5921</td>\n",
       "      <td>rapid_renal_clearance</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6000</td>\n",
       "      <td>ar42j_cell</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6105</td>\n",
       "      <td>bombesin_bn</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6106</td>\n",
       "      <td>bn</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6114</td>\n",
       "      <td>bnrs</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6115</td>\n",
       "      <td>gastrin_release_peptide</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            word  class\n",
       "409                        tumor      7\n",
       "877          intensity_histogram      7\n",
       "881                        image      7\n",
       "1000                      tumour      7\n",
       "1742                         111      7\n",
       "2399                        sst2      7\n",
       "2661             carcinoid_tumor      7\n",
       "2667                     imaging      7\n",
       "2674  bronchopulmonary_carcinoid      7\n",
       "2679                   125ityr10      7\n",
       "3715             high_resolution      7\n",
       "3763           misinterpretation      7\n",
       "4149                      uptake      7\n",
       "4251               semiautomated      7\n",
       "5921       rapid_renal_clearance      7\n",
       "6000                  ar42j_cell      7\n",
       "6105                 bombesin_bn      7\n",
       "6106                          bn      7\n",
       "6114                        bnrs      7\n",
       "6115     gastrin_release_peptide      7"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[df['class'] == 7].head(20) # no-use"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>word</th>\n",
       "      <th>class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>415</td>\n",
       "      <td>cycle</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1042</td>\n",
       "      <td>steroid_hormone</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1090</td>\n",
       "      <td>kisspeptin</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1091</td>\n",
       "      <td>pregnancy</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1098</td>\n",
       "      <td>kiss1</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1101</td>\n",
       "      <td>metastin</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1104</td>\n",
       "      <td>inactivate_mutation</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1105</td>\n",
       "      <td>gpr54_kiss1r</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1108</td>\n",
       "      <td>puberty</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1111</td>\n",
       "      <td>kisspeptin_kiss1r_system</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1113</td>\n",
       "      <td>as_an_essential_gatekeeper</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1114</td>\n",
       "      <td>reproductive_function</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1118</td>\n",
       "      <td>gonadotropic_axis</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1119</td>\n",
       "      <td>placentation</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1122</td>\n",
       "      <td>kisspeptin_kiss1r_signal</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1258</td>\n",
       "      <td>corticotrophin_release_hormone</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1260</td>\n",
       "      <td>thyroid_gland</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1263</td>\n",
       "      <td>crh</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1265</td>\n",
       "      <td>crhr1</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1267</td>\n",
       "      <td>ucn_crhr1</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                word  class\n",
       "415                            cycle      8\n",
       "1042                 steroid_hormone      8\n",
       "1090                      kisspeptin      8\n",
       "1091                       pregnancy      8\n",
       "1098                           kiss1      8\n",
       "1101                        metastin      8\n",
       "1104             inactivate_mutation      8\n",
       "1105                    gpr54_kiss1r      8\n",
       "1108                         puberty      8\n",
       "1111        kisspeptin_kiss1r_system      8\n",
       "1113      as_an_essential_gatekeeper      8\n",
       "1114           reproductive_function      8\n",
       "1118               gonadotropic_axis      8\n",
       "1119                    placentation      8\n",
       "1122        kisspeptin_kiss1r_signal      8\n",
       "1258  corticotrophin_release_hormone      8\n",
       "1260                   thyroid_gland      8\n",
       "1263                             crh      8\n",
       "1265                           crhr1      8\n",
       "1267                       ucn_crhr1      8"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[df['class'] == 8].head(20) # hormone"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>word</th>\n",
       "      <th>class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>know_about</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>well_appreciate</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>endocannabinoid_system</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>34</td>\n",
       "      <td>endocannabinoid</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>38</td>\n",
       "      <td>plant_derive</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>39</td>\n",
       "      <td>synthetic_cannabinoid</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>50</td>\n",
       "      <td>biological_process</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>58</td>\n",
       "      <td>greatly_expand</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>59</td>\n",
       "      <td>our_knowledge</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>60</td>\n",
       "      <td>how</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>65</td>\n",
       "      <td>health</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>66</td>\n",
       "      <td>disease</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74</td>\n",
       "      <td>here_we_review</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>75</td>\n",
       "      <td>with_an_emphasis_on</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>79</td>\n",
       "      <td>copyright_2014</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>80</td>\n",
       "      <td>publish_by_elsevi_ltd</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>84</td>\n",
       "      <td>expand</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>85</td>\n",
       "      <td>pharmacological_toolbox</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>90</td>\n",
       "      <td>organism</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>96</td>\n",
       "      <td>physiological_process</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                       word  class\n",
       "4                know_about      9\n",
       "13          well_appreciate      9\n",
       "15   endocannabinoid_system      9\n",
       "34          endocannabinoid      9\n",
       "38             plant_derive      9\n",
       "39    synthetic_cannabinoid      9\n",
       "50       biological_process      9\n",
       "58           greatly_expand      9\n",
       "59            our_knowledge      9\n",
       "60                      how      9\n",
       "65                   health      9\n",
       "66                  disease      9\n",
       "74           here_we_review      9\n",
       "75      with_an_emphasis_on      9\n",
       "79           copyright_2014      9\n",
       "80    publish_by_elsevi_ltd      9\n",
       "84                   expand      9\n",
       "85  pharmacological_toolbox      9\n",
       "90                 organism      9\n",
       "96    physiological_process      9"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[df['class'] == 9].head(20) # no use"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>word</th>\n",
       "      <th>class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>88</td>\n",
       "      <td>genome</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>91</td>\n",
       "      <td>encode</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>187</td>\n",
       "      <td>an_orphan_member</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>196</td>\n",
       "      <td>ortholog</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>247</td>\n",
       "      <td>gene</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>363</td>\n",
       "      <td>foundation_for_understand</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>385</td>\n",
       "      <td>orphan_gpcr</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>465</td>\n",
       "      <td>hitherto_unknown</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>489</td>\n",
       "      <td>rmc1</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>608</td>\n",
       "      <td>module</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>696</td>\n",
       "      <td>percept</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>718</td>\n",
       "      <td>perceptually</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>813</td>\n",
       "      <td>fly</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>817</td>\n",
       "      <td>drosophila_melanogaster</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>847</td>\n",
       "      <td>homology_between</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>848</td>\n",
       "      <td>drosophila</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>869</td>\n",
       "      <td>custom</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1438</td>\n",
       "      <td>subdivide_into_two</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1440</td>\n",
       "      <td>subfamily</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1465</td>\n",
       "      <td>intron</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           word  class\n",
       "88                       genome     10\n",
       "91                       encode     10\n",
       "187            an_orphan_member     10\n",
       "196                    ortholog     10\n",
       "247                        gene     10\n",
       "363   foundation_for_understand     10\n",
       "385                 orphan_gpcr     10\n",
       "465            hitherto_unknown     10\n",
       "489                        rmc1     10\n",
       "608                      module     10\n",
       "696                     percept     10\n",
       "718                perceptually     10\n",
       "813                         fly     10\n",
       "817     drosophila_melanogaster     10\n",
       "847            homology_between     10\n",
       "848                  drosophila     10\n",
       "869                      custom     10\n",
       "1438         subdivide_into_two     10\n",
       "1440                  subfamily     10\n",
       "1465                     intron     10"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[df['class'] == 10].head(20) # genetics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>word</th>\n",
       "      <th>class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>103</td>\n",
       "      <td>prescribe</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>152</td>\n",
       "      <td>cholestatic_pruritus</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>161</td>\n",
       "      <td>experience</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>168</td>\n",
       "      <td>patient</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>180</td>\n",
       "      <td>therapy</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>328</td>\n",
       "      <td>card</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>444</td>\n",
       "      <td>hm74a</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>450</td>\n",
       "      <td>niacin</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>740</td>\n",
       "      <td>oral</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>866</td>\n",
       "      <td>pharmacodynamic</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1016</td>\n",
       "      <td>clinical_datum</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1109</td>\n",
       "      <td>man</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1207</td>\n",
       "      <td>active_metabolite</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1459</td>\n",
       "      <td>bleed_time</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1460</td>\n",
       "      <td>p2y_12</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1549</td>\n",
       "      <td>clinical</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1669</td>\n",
       "      <td>mk6096</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1671</td>\n",
       "      <td>insomnia</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1694</td>\n",
       "      <td>plasma_concentration</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1715</td>\n",
       "      <td>care</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      word  class\n",
       "103              prescribe     11\n",
       "152   cholestatic_pruritus     11\n",
       "161             experience     11\n",
       "168                patient     11\n",
       "180                therapy     11\n",
       "328                   card     11\n",
       "444                  hm74a     11\n",
       "450                 niacin     11\n",
       "740                   oral     11\n",
       "866        pharmacodynamic     11\n",
       "1016        clinical_datum     11\n",
       "1109                   man     11\n",
       "1207     active_metabolite     11\n",
       "1459            bleed_time     11\n",
       "1460                p2y_12     11\n",
       "1549              clinical     11\n",
       "1669                mk6096     11\n",
       "1671              insomnia     11\n",
       "1694  plasma_concentration     11\n",
       "1715                  care     11"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[df['class'] == 11].head(20) # pharma"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>word</th>\n",
       "      <th>class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>immune_response</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>62</td>\n",
       "      <td>immune</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>78</td>\n",
       "      <td>immune_system</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>169</td>\n",
       "      <td>sera</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>186</td>\n",
       "      <td>x4</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>202</td>\n",
       "      <td>humanized_mouse</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>244</td>\n",
       "      <td>rhesus_macaque</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>252</td>\n",
       "      <td>macaque</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>443</td>\n",
       "      <td>puma_g</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>520</td>\n",
       "      <td>pathogen</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>545</td>\n",
       "      <td>microbiota</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>546</td>\n",
       "      <td>bifidobacterium</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>574</td>\n",
       "      <td>germ_free_mouse</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>618</td>\n",
       "      <td>enumeration</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>639</td>\n",
       "      <td>highly_efficient</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>994</td>\n",
       "      <td>expansion</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1035</td>\n",
       "      <td>durable</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1327</td>\n",
       "      <td>immune_reaction</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1427</td>\n",
       "      <td>neutralize</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1453</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  word  class\n",
       "18     immune_response     12\n",
       "62              immune     12\n",
       "78       immune_system     12\n",
       "169               sera     12\n",
       "186                 x4     12\n",
       "202    humanized_mouse     12\n",
       "244     rhesus_macaque     12\n",
       "252            macaque     12\n",
       "443             puma_g     12\n",
       "520           pathogen     12\n",
       "545         microbiota     12\n",
       "546    bifidobacterium     12\n",
       "574    germ_free_mouse     12\n",
       "618        enumeration     12\n",
       "639   highly_efficient     12\n",
       "994          expansion     12\n",
       "1035           durable     12\n",
       "1327   immune_reaction     12\n",
       "1427        neutralize     12\n",
       "1453         phenotype     12"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[df['class'] == 12].head(20) # immune"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>word</th>\n",
       "      <th>class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>via</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>178</td>\n",
       "      <td>ba</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>179</td>\n",
       "      <td>modulating</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>269</td>\n",
       "      <td>without_affect</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>283</td>\n",
       "      <td>wash_out</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>299</td>\n",
       "      <td>spontaneous</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>311</td>\n",
       "      <td>mechanism</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>340</td>\n",
       "      <td>release</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>356</td>\n",
       "      <td>distal</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>397</td>\n",
       "      <td>crc</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>433</td>\n",
       "      <td>conclusion</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>488</td>\n",
       "      <td>ganglion</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>537</td>\n",
       "      <td>produce</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>650</td>\n",
       "      <td>alpha1a</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>711</td>\n",
       "      <td>response</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>734</td>\n",
       "      <td>unmasked</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>749</td>\n",
       "      <td>stimulus</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>772</td>\n",
       "      <td>vascular</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>790</td>\n",
       "      <td>cause</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>910</td>\n",
       "      <td>mediate</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               word  class\n",
       "19              via     13\n",
       "178              ba     13\n",
       "179      modulating     13\n",
       "269  without_affect     13\n",
       "283        wash_out     13\n",
       "299     spontaneous     13\n",
       "311       mechanism     13\n",
       "340         release     13\n",
       "356          distal     13\n",
       "397             crc     13\n",
       "433      conclusion     13\n",
       "488        ganglion     13\n",
       "537         produce     13\n",
       "650         alpha1a     13\n",
       "711        response     13\n",
       "734        unmasked     13\n",
       "749        stimulus     13\n",
       "772        vascular     13\n",
       "790           cause     13\n",
       "910         mediate     13"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[df['class'] == 13].head(20) # no-use"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>word</th>\n",
       "      <th>class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>cannabinoid</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>cb2</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>154</td>\n",
       "      <td>cholestasis</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>163</td>\n",
       "      <td>chronic_itch</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>165</td>\n",
       "      <td>pruritogen</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>167</td>\n",
       "      <td>cholestatic</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>173</td>\n",
       "      <td>injection</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>176</td>\n",
       "      <td>itch</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>195</td>\n",
       "      <td>mrgpr</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>209</td>\n",
       "      <td>scratch</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>266</td>\n",
       "      <td>dose</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>428</td>\n",
       "      <td>markedly_suppress</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>722</td>\n",
       "      <td>intensity</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>756</td>\n",
       "      <td>sensation</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1160</td>\n",
       "      <td>deliver</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1256</td>\n",
       "      <td>v</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1298</td>\n",
       "      <td>t</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1450</td>\n",
       "      <td>pain</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1750</td>\n",
       "      <td>last</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1765</td>\n",
       "      <td>skin</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   word  class\n",
       "6           cannabinoid     14\n",
       "9                   cb2     14\n",
       "154         cholestasis     14\n",
       "163        chronic_itch     14\n",
       "165          pruritogen     14\n",
       "167         cholestatic     14\n",
       "173           injection     14\n",
       "176                itch     14\n",
       "195               mrgpr     14\n",
       "209             scratch     14\n",
       "266                dose     14\n",
       "428   markedly_suppress     14\n",
       "722           intensity     14\n",
       "756           sensation     14\n",
       "1160            deliver     14\n",
       "1256                  v     14\n",
       "1298                  t     14\n",
       "1450               pain     14\n",
       "1750               last     14\n",
       "1765               skin     14"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[df['class'] == 14].head(20) # pain"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>word</th>\n",
       "      <th>class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>151</td>\n",
       "      <td>may_contribute</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>249</td>\n",
       "      <td>downregulate</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>285</td>\n",
       "      <td>during</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>296</td>\n",
       "      <td>prevention</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>342</td>\n",
       "      <td>tilt</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>453</td>\n",
       "      <td>hb</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>457</td>\n",
       "      <td>circulation</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>467</td>\n",
       "      <td>circulate</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>468</td>\n",
       "      <td>diabete</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>507</td>\n",
       "      <td>conclusion_these_datum</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>511</td>\n",
       "      <td>blood</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>783</td>\n",
       "      <td>ist</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>807</td>\n",
       "      <td>progression</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>844</td>\n",
       "      <td>map</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>952</td>\n",
       "      <td>ra</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>978</td>\n",
       "      <td>ramp1</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>993</td>\n",
       "      <td>death</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>996</td>\n",
       "      <td>endothelin1_et1</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1006</td>\n",
       "      <td>et1</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1039</td>\n",
       "      <td>beneficial_effect</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        word  class\n",
       "151           may_contribute     15\n",
       "249             downregulate     15\n",
       "285                   during     15\n",
       "296               prevention     15\n",
       "342                     tilt     15\n",
       "453                       hb     15\n",
       "457              circulation     15\n",
       "467                circulate     15\n",
       "468                  diabete     15\n",
       "507   conclusion_these_datum     15\n",
       "511                    blood     15\n",
       "783                      ist     15\n",
       "807              progression     15\n",
       "844                      map     15\n",
       "952                       ra     15\n",
       "978                    ramp1     15\n",
       "993                    death     15\n",
       "996          endothelin1_et1     15\n",
       "1006                     et1     15\n",
       "1039       beneficial_effect     15"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[df['class'] == 15].head(20) # not-obvious"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>word</th>\n",
       "      <th>class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>69</td>\n",
       "      <td>genetic</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>148</td>\n",
       "      <td>mrgprx4</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>230</td>\n",
       "      <td>socioeconomic</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>235</td>\n",
       "      <td>aud</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>242</td>\n",
       "      <td>dna_methylome</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>250</td>\n",
       "      <td>heavy_alcohol</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>544</td>\n",
       "      <td>healthy</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>629</td>\n",
       "      <td>weighted</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>654</td>\n",
       "      <td>dud</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>703</td>\n",
       "      <td>psychophysically</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>730</td>\n",
       "      <td>allelic_variation</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>745</td>\n",
       "      <td>genotype</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>782</td>\n",
       "      <td>ren</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1049</td>\n",
       "      <td>vdr</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1266</td>\n",
       "      <td>crhr2</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1468</td>\n",
       "      <td>susceptibility</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1476</td>\n",
       "      <td>exon</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1492</td>\n",
       "      <td>frameshift_mutation</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1534</td>\n",
       "      <td>overrepresente_among</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1836</td>\n",
       "      <td>pcr</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      word  class\n",
       "69                 genetic     16\n",
       "148                mrgprx4     16\n",
       "230          socioeconomic     16\n",
       "235                    aud     16\n",
       "242          dna_methylome     16\n",
       "250          heavy_alcohol     16\n",
       "544                healthy     16\n",
       "629               weighted     16\n",
       "654                    dud     16\n",
       "703       psychophysically     16\n",
       "730      allelic_variation     16\n",
       "745               genotype     16\n",
       "782                    ren     16\n",
       "1049                   vdr     16\n",
       "1266                 crhr2     16\n",
       "1468        susceptibility     16\n",
       "1476                  exon     16\n",
       "1492   frameshift_mutation     16\n",
       "1534  overrepresente_among     16\n",
       "1836                   pcr     16"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[df['class'] == 16].head(20) # genetic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>word</th>\n",
       "      <th>class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>55</td>\n",
       "      <td>generation</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>132</td>\n",
       "      <td>activate</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>155</td>\n",
       "      <td>slowing</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>200</td>\n",
       "      <td>activation</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>357</td>\n",
       "      <td>nucleotide</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>367</td>\n",
       "      <td>slow</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>427</td>\n",
       "      <td>depletion</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>498</td>\n",
       "      <td>polarize</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>499</td>\n",
       "      <td>basolateral_membrane</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>589</td>\n",
       "      <td>application</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>733</td>\n",
       "      <td>pathway</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>880</td>\n",
       "      <td>fluorescence_microscopy</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>928</td>\n",
       "      <td>transient</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>935</td>\n",
       "      <td>rapid_transient</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>938</td>\n",
       "      <td>pertussis_toxin_sensitive</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>939</td>\n",
       "      <td>mimic</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1019</td>\n",
       "      <td>trigger</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1164</td>\n",
       "      <td>channel</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1180</td>\n",
       "      <td>bypass</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1230</td>\n",
       "      <td>ca2</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           word  class\n",
       "55                   generation     17\n",
       "132                    activate     17\n",
       "155                     slowing     17\n",
       "200                  activation     17\n",
       "357                  nucleotide     17\n",
       "367                        slow     17\n",
       "427                   depletion     17\n",
       "498                    polarize     17\n",
       "499        basolateral_membrane     17\n",
       "589                 application     17\n",
       "733                     pathway     17\n",
       "880     fluorescence_microscopy     17\n",
       "928                   transient     17\n",
       "935             rapid_transient     17\n",
       "938   pertussis_toxin_sensitive     17\n",
       "939                       mimic     17\n",
       "1019                    trigger     17\n",
       "1164                    channel     17\n",
       "1180                     bypass     17\n",
       "1230                        ca2     17"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[df['class'] == 17].head(20) # signaling"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>word</th>\n",
       "      <th>class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>98</td>\n",
       "      <td>drug</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>140</td>\n",
       "      <td>challenge</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>172</td>\n",
       "      <td>acute</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>208</td>\n",
       "      <td>animal</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>217</td>\n",
       "      <td>alleviate</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>221</td>\n",
       "      <td>alcohol</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>225</td>\n",
       "      <td>ethanol_intake</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>227</td>\n",
       "      <td>chronic</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>243</td>\n",
       "      <td>nucleus_accumben_core_nacc</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>251</td>\n",
       "      <td>drinking</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>260</td>\n",
       "      <td>finding_suggest_that</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>262</td>\n",
       "      <td>reduce_alcohol_intake</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>264</td>\n",
       "      <td>two_bottle_choice</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>265</td>\n",
       "      <td>5_mg_kg</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>270</td>\n",
       "      <td>total_fluid_intake</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>271</td>\n",
       "      <td>locomotor_activity</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>272</td>\n",
       "      <td>saccharin_preference</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>274</td>\n",
       "      <td>repeat</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>276</td>\n",
       "      <td>ethanol</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>277</td>\n",
       "      <td>escalation</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           word  class\n",
       "98                         drug     18\n",
       "140                   challenge     18\n",
       "172                       acute     18\n",
       "208                      animal     18\n",
       "217                   alleviate     18\n",
       "221                     alcohol     18\n",
       "225              ethanol_intake     18\n",
       "227                     chronic     18\n",
       "243  nucleus_accumben_core_nacc     18\n",
       "251                    drinking     18\n",
       "260        finding_suggest_that     18\n",
       "262       reduce_alcohol_intake     18\n",
       "264           two_bottle_choice     18\n",
       "265                     5_mg_kg     18\n",
       "270          total_fluid_intake     18\n",
       "271          locomotor_activity     18\n",
       "272        saccharin_preference     18\n",
       "274                      repeat     18\n",
       "276                     ethanol     18\n",
       "277                  escalation     18"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[df['class'] == 18].head(20) # drug"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>word</th>\n",
       "      <th>class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>56</td>\n",
       "      <td>mouse_deficient</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>166</td>\n",
       "      <td>up_regulated</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>185</td>\n",
       "      <td>mas_relate_g_protein_couple</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>267</td>\n",
       "      <td>tc_g_1008</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>413</td>\n",
       "      <td>apoptosis</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>414</td>\n",
       "      <td>flow_cytometry</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>417</td>\n",
       "      <td>migration</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>423</td>\n",
       "      <td>upregulate</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>470</td>\n",
       "      <td>retinopathy</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>477</td>\n",
       "      <td>diabetic_retinopathy</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>492</td>\n",
       "      <td>arpe19_cell</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>502</td>\n",
       "      <td>arpe19</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>515</td>\n",
       "      <td>inflammatory_response</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>517</td>\n",
       "      <td>chemoattractant</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>521</td>\n",
       "      <td>pattern_recognition</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>524</td>\n",
       "      <td>toll_like</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>525</td>\n",
       "      <td>tlrs</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>533</td>\n",
       "      <td>inflammatory_disease</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>541</td>\n",
       "      <td>ulcerative_colitis</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>543</td>\n",
       "      <td>colitic</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            word  class\n",
       "56               mouse_deficient     19\n",
       "166                 up_regulated     19\n",
       "185  mas_relate_g_protein_couple     19\n",
       "267                    tc_g_1008     19\n",
       "413                    apoptosis     19\n",
       "414               flow_cytometry     19\n",
       "417                    migration     19\n",
       "423                   upregulate     19\n",
       "470                  retinopathy     19\n",
       "477         diabetic_retinopathy     19\n",
       "492                  arpe19_cell     19\n",
       "502                       arpe19     19\n",
       "515        inflammatory_response     19\n",
       "517              chemoattractant     19\n",
       "521          pattern_recognition     19\n",
       "524                    toll_like     19\n",
       "525                         tlrs     19\n",
       "533         inflammatory_disease     19\n",
       "541           ulcerative_colitis     19\n",
       "543                      colitic     19"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[df['class'] == 19].head(20) # phenotype / functionality"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### playground for 20 clusters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>what</th>\n",
       "      <th>4</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>we</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>know</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>and</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>do_not</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>know_about</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         what   4\n",
       "0          we  32\n",
       "1        know  32\n",
       "2         and  32\n",
       "3      do_not  32\n",
       "4  know_about   4"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df35 = pd.read_csv('word_class35.csv')\n",
    "df35.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "df35.columns = ['word','label']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cluster:0 ['intensity_histogram', 'image', 'tumour', 'conjugate', '111', 'sst2', 'carcinoid_tumor', 'somatostatin', 'imaging', '125ityr10', 'radiation', 'misinterpretation', 'uptake', 'rapid_renal_clearance', 'ar42j_cell', 'bombesin_bn', 'bn', 'bnrs', 'gastrin_release_peptide', 'grpr']\n",
      "\n",
      "\n",
      "cluster:1 ['screening_assay', 'allosteric_ligand', 'topographically_distinct_site', 'simulation', 'unbind', 'computational_method', 'metadynamic', 'markov_state_model', 'msms', 'correlated', 'motion', 'gpcr', 'unbinding', 'slow_off_rate', 'resolution', 'pharmacophore_base_virtual_screening', 'virtual_screening', 'cost_effective', 'rational_drug_design', 'spatial_configuration']\n",
      "\n",
      "\n",
      "cluster:2 ['gpr56', 'rheostat', 'catenin', 'enpp2', 'secreted_lysophospholipase_d', 'architecture', 'organize_into', 'smoothen', 'hedgehog_hh_signal_play', 'conditionally', 'hh_signal', 'hh_pathway', 'hh_pathway_component', 'smo', 'cre_transgenic_mouse', 'gli2', 'gli3', 'lens_epithelium', 'mitotic', 'conditional_deletion']\n",
      "\n",
      "\n",
      "cluster:3 ['ortholog', 'foundation_for_understand', 'taste', 'percept', 'perceptually', 'fly', 'drosophila_melanogaster', 'drosophila', 'subfamily', 'olfactory', 'large_family', 'olfactory_epithelium', 'zebrafish', 'insect', 'chemosensory', 'pkd1l3', 'sour_taste', 'diversity', 'transcriptomic', 'mammal']\n",
      "\n",
      "\n",
      "cluster:4 ['know_about', 'well_appreciate', 'endocannabinoid_system', 'endocannabinoid', 'plant_derive', 'synthetic_cannabinoid', 'biological_process', 'mouse_deficient', 'greatly_expand', 'our_knowledge', 'how', 'health', 'disease', 'here_we_review', 'with_an_emphasis_on', 'copyright_2014', 'publish_by_elsevi_ltd', 'expand', 'pharmacological_toolbox', 'organism']\n",
      "\n",
      "\n",
      "cluster:5 ['dopamine', 'dopamine_da', 'motor_function', 'da', 'da_neurotransmission', 'dopaminergic_neuron', 'dopaminergic_system', 'clozapine_metabolite_n_desmethylclozapine', 'n_desmethylclozapine_ndmc', 'antipsychotic_clozapine', 'ndmc', 'clozapine', 'dopamine_d2', 'd2dependent', 'd2r', 'd2', 'd1d3', 'd1', 'd3', 'dopamine_d3']\n",
      "\n",
      "\n",
      "cluster:6 ['genetic', 'retinopathy', 'homozygous_deletion', 'vdr', 'oca', 'inactivate_mutation', 'phenotype', 'frameshift_mutation', 'ocular', 'mutation', 'defect', 'humphrey', 'casr', 'polycystin', 'autosomal_dominant_polycystic_kidney_disease', 'proteolytic_site_gps', 'gps', 'dysplasia', 'bilateral_frontoparietal_polymicrogyria', 'phenotypic_consequence']\n",
      "\n",
      "\n",
      "cluster:7 ['g_protein_couple', 'zn_2', 'together_our_finding_suggest_that', 'heterotrimeric_g_protein', 'signaling_cascade', 'pi3k_akt_signal', 'overexpression', 'here_we_show_that', 'extracellular_signal_regulated_kinase_12', 'gq', 'g1213', 'gi_protein', 'intracellular_signal_pathway', 'tyrosine_kinase', 'transactivation', 'subcellular_compartment', 'c_src_tyrosine_kinase', 'gbetagamma_subunit', 'mitogen_activate_protein_kinase', 'pertussis_toxin_sensitive_heterotrimeric_g_protein']\n",
      "\n",
      "\n",
      "cluster:8 ['immune_response', 'immune', 'immune_system', 'flow_cytometry', 'migration', 'pathogen', 'pattern_recognition', 'toll_like', 'tlrs', 'colitic', 'microbiota', 'bifidobacterium', 'recruitment', 'germ_free_mouse', 'expansion', 'immune_reaction', 'cd39', 't_cell', 'c_maf', 'autoimmune']\n",
      "\n",
      "\n",
      "cluster:9 ['up_regulated', 'alleviate', 'downregulate', 'tc_g_1008', 'bdnf', 'survival', 'apoptosis', 'upregulate', 'diabetic_retinopathy', 'cell_line_arpe19', 'rgc5', 'rgc5_cell', 'protect_against', 'exacerbated', 'ko_mouse', 'ko_mouse_display', 'severely', 'death', 'beneficial_effect', 'abate']\n",
      "\n",
      "\n",
      "cluster:10 ['genome', 'encode', 'an_orphan_member', 'expression', 'gene', 'construct', 'translation', 'orphan_gpcr', 'family', 'mrna', 'hitherto_unknown', 'rt_pcr', 'rmc1', 'element', 'decoy', 'downstream', 'homology_between', 'nuclear', 'panel', 'parafollicular']\n",
      "\n",
      "\n",
      "cluster:11 ['beta2adrenergic', 'beta2ar', 'intercept', 'skeletal_muscle', 'myocardium', 'cardiac', 'heart', 'hf', 'rv', 'contractility', 'myofilament_ca2_sensitivity', 'beta_adrenoceptor_beta_ars', 'myocardial', 'beta_ar', 'beta1ar', 'systolic', 'cardiac_hypertrophy', 'left_ventricular', 'cardiac_function', 'leave_ventricle']\n",
      "\n",
      "\n",
      "cluster:12 ['diabete', 'progression', 'ras', 'modify_protein_ramp', 'ramp1', 'endothelin1_et1', 'et1', 'fibrosis', 'endothelial_dysfunction', 'vascular_remodeling', 'fibrotic_process', 'pulmonary_arterial_hypertension_pah', 'monocrotaline_induce', 'pah', 'salt_sensitive_hypertension', 'sodium_homeostasis', 'normotension', 'natriuretic', 'volume_expansion', 'sodium_load']\n",
      "\n",
      "\n",
      "cluster:13 ['arpe19_cell', 'arpe19', 'inflammatory_response', 'chemoattractant', 'inflammatory_disease', 'ulcerative_colitis', 'colitis', 'asthma', 'inflammatory_mediator', 'bacteria', 'bacterial', 'murine_model', 'polysaccharide', 'protease_activate_receptor1', 'protease_activate', 'par', 'hemostasis', 'thrombosis', 'par1', 'mesothelial_cell']\n",
      "\n",
      "\n",
      "cluster:14 ['cycle', 'steroid_hormone', 'kisspeptin', 'pregnancy', 'kiss1', 'gpr54_kiss1r', 'puberty', 'kisspeptin_kiss1r_system', 'as_an_essential_gatekeeper', 'reproductive_function', 'gonadotropic_axis', 'placentation', 'kisspeptin_kiss1r_signal', 'corticotrophin_release_hormone', 'thyroid_gland', 'crh', 'crhr2', 'ucn_crhr1', 'crhr1ir', '2013_blackwell_verlag_gmbh']\n",
      "\n",
      "\n",
      "cluster:15 ['dark', 'illuminate', 'arrestin', 'retinal_pigment_epithelium', 'retinal', 'retina', 'rpe', 'mller', 'viscosity', 'spectrum', 'lens', 'ophthalmol', 'vis', 'transition', 'extraocular', 'vitamin', 'proteobacteria', 'eye', 'visual', 'long_live']\n",
      "\n",
      "\n",
      "cluster:16 ['prescribe', 'cholestatic_pruritus', 'experience', 'patient', 'therapy', 'card', 'niacin', 'pharmacodynamic', 'active_metabolite', 'bleed_time', 'p2y_12', 'clinical', 'mk6096', 'insomnia', 'care', 'committee', 'expert', 'clinical_efficacy', 'side_effect', 'titrate']\n",
      "\n",
      "\n",
      "cluster:17 ['metabotropic', 'excitatory', 'inhibitory_neurotransmission', 'glutamate_release', 'application', 'input', 'mso', 'ramp', 'g_protein_couple_inward_rectifi_potassium_channel', 'girk_current', 'girk', 'gabab', 'kctd12', 'electrophysiological_approach', 'kctd12induce', 'neuroscience_2012_federation', 'european_neuroscience_society', 'blackwell_publishing_ltd', 'input_output_relationship', 'plasticity']\n",
      "\n",
      "\n",
      "cluster:18 ['cannabinoid', 'cb2', 'cholestasis', 'prototypical', '35sgtps_bind', 'cannabinoid_cb1', 'cb1', 't', 'nociceptin_orphanin_fq', 'nop', 'micro', 'morphine', 'opioid', 'd_tyr', 'endogenous_ligand', 'tyr_d', 'd_trp', 'diabetic_mouse', 'mu', 'win55212_2']\n",
      "\n",
      "\n",
      "cluster:19 ['generation', 'ca_2_imaging', 'activation', 'nucleotide', 'polarize', 'basolateral_membrane', 'fluorescence_microscopy', 'transient', 'rapid_transient', 'trigger', 'channel', 'bypass', 'ca2', 'liberation', 'xenopus_oocyte', 'current', 'a2b_adenosine', 'k_channel', 'k_current', 'p2']\n",
      "\n",
      "\n",
      "cluster:20 ['x4', 'humanized_mouse', 'rhesus_macaque', 'macaque', 'productive', 'hiv_coreceptor', 'gpr15', 'alternative_coreceptor', 'siv', 'tropism', 'siv_isolate', 'coreceptor', 'efficient_entry_into', 't_cell_line', 'cemx174', 'siv_infection', 'cemx174_cell', 'peripheral_blood_mononuclear_cell', 'pbmcs', 'efficiently_infect']\n",
      "\n",
      "\n",
      "cluster:21 ['along', 'intense', 'numerous', 'neuron', 'immunohistochemistry', 'distribution', 'localization', 'situ_hybridization', 'immunofluorescent_technique', 'restrict', 'where_it', 'subset', 'somatosensory', 'give_rise', 'distinguishable', 'region', 'innervate', 'confocal_laser_scan_microscopy', 'immunostaining', 'coincide_with']\n",
      "\n",
      "\n",
      "cluster:22 ['orthosteric_site', 'orthosteric_ligand', 'key_step', 'compound', 'pharmacophore', 'bind_mode', 'evaluation', 'chain_length', 'accommodate', 'functional_assay', 'weak_partial_agonist', 'o', 'parent', 'new_class', 'position', 'orally_bioavailable', 'potent', 'structurally_distinct_class', 'pharmacokinetic_property', 'an_effort']\n",
      "\n",
      "\n",
      "cluster:23 ['ba', 'wash_out', 'crc', 'ganglion', 'alpha1a', 'unmasked', 'vascular', 'mediate', 'elicit', 'more_effective', 'evoke', 'smooth_muscle', 'blood_vessel', 'prostaglandin', 'endow_with', 'platelet_aggregatory', 'neurogenic', 'smooth_muscle_contraction', 'airway', 'intestine']\n",
      "\n",
      "\n",
      "cluster:24 ['alcohol_use_disorder_aud', 'socioeconomic', 'aud', 'dna_methylome', 'heavy_alcohol', 'psychophysically', 'allelic_variation', 'psychiatric_disorder', 'crhr1', 'overrepresente_among', 'neurogenetic', 'cognitive_function', 'work_memory_wm', 'repetition', 'wm', 'comt', 'catechol_o_methyl_transferase', 'score', 'semantic', 'indexing']\n",
      "\n",
      "\n",
      "cluster:25 ['chronic_itch', 'pruritogen', 'cholestatic', 'itch', 'sensory_neuron', 'mas_relate_g_protein_couple', 'mrgpr_family', 'mrgpr', 'scratch', 'cholestatic_itch', 'inflammation', 'arthritis', 'sensation', 'sensory_processing', 'pain', 'visceral_perception', 'pruritus', 'polymodal', 'chronic_pain', 'pin']\n",
      "\n",
      "\n",
      "cluster:26 ['drug', 'rodent', 'behaviour', 'behavioural', 'serotonin_5hta', 'norclozapine', 'serotonin_5ht1a', 'partial_agonist_property', 'post_synaptic', 'sleep', 'sleep_promote', 'agonistic_action', 'trace_amine', 'phenylethylamine_pea', 'aminergic', 'pca', 'purported', 'neurotransmission', 'behavioral_despair', 'corticosterone']\n",
      "\n",
      "\n",
      "cluster:27 ['control', 'contribute', 'role', 'increase', 'challenge', 'may_contribute', 'slowing', 'acute', 'injection', 'modulating', 'animal', 'chronic', 'prior', 'aim', 'after', 'balance', 'finding_suggest_that', '5_mg_kg', 'dose', 'reduce']\n",
      "\n",
      "\n",
      "cluster:28 ['hypermethylate', 'colorectal_cancer', 'metastasis', 'epithelialmesenchymal_transition', 'g_proteincoupled', '56_gpr56', 'oncogene', 'malignancy', 'colorectal_cancer_crc', 'quantitative_real_time', 'qrtpcr', 'prognostic_significance', 'tumor', 'cell_count_kit8', 'transwell_assay', 'predict_poor_prognosis', 'proliferation_migration', 'invasion', 'crc_progression', 'crc_metastasis']\n",
      "\n",
      "\n",
      "cluster:29 ['alcohol', 'ethanol_intake', 'nucleus_accumben_core_nacc', 'drinking', 'reduce_alcohol_intake', 'two_bottle_choice', 'total_fluid_intake', 'locomotor_activity', 'saccharin_preference', 'ethanol', 'escalation', 'intermittent_access', '2bc', 'paradigm', 'nacc', 'access', 'intake', 'exploration', 'performance', 'maltodextrin']\n",
      "\n",
      "\n",
      "cluster:30 ['hematopoietic', 'proliferation', 'anti_fibrotic_effect', 'autotaxin_atx', 'atx', 'an_autocrine_paracrine', 'p57kip2', 'corneal_endothelial', 'mesenchymal_cell', 'stem_cell', 'stroma', 'real_time_reverse_transcription_polymerase_chain_reaction', 'stromal', 'nuclear_factor_b', 'bone', 'proliferate', 'adrb12', 'angiogenesis', 'mural_cell', 'adherent']\n",
      "\n",
      "\n",
      "cluster:31 ['bile_acid', 'bile_flow', 'bile_acid_bas', 'bas', 'gpr39', '39_gpr39', 'gpr109a', 'puma_g', 'hm74a', 'beta_hydroxybutyrate', 'ketone_body', 'an_important_energy_source', 'gut_microbiota', 'gpr43', 'intestinal_microbiota', 'short_chain_fatty_acid_scfas', 'fermentation', 'dietary_fibre', 'scfas', 'fermentable']\n",
      "\n",
      "\n",
      "cluster:32 ['we', 'know', 'and', 'do_not', 'the', 'receptor', '2', 'it', 'have', 'be', 'that', 'can', 'regulate', 'via', 'which', 'primarily', 'express', 'by', 'cell', 'of']\n",
      "\n",
      "\n",
      "cluster:33 ['mrgprx4', 'healthy', 'genotype', 'ren', 'susceptibility', '142', 'serum_triglyceride_level', 'ld', 'variant', 'genetic_polymorphism', '2012_american', 'individual_homozygous_for', 'allele', 'adjust', '8_kg_m2', '50_year', 'risk', 'adrb1', 'adrb2', 'n29']\n",
      "\n",
      "\n",
      "cluster:34 ['7tm', 'seven_transmembrane_7tm_g_protein_couple', 'domain', 'gpcrs', 'allosteric_communication_between', 'structural', 'helix', 'strand', 'conformational_change', 'disordering', 'mutant', 'recognition', 'heterodimer', 'wild_type', 'plasma_membrane_localization', 'homodimerization', 'complex_formation', 'cos7_cell', 'c_terminal', 'tyrosine_residue']\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for i in range(35):\n",
    "    print('cluster:{}'.format(i), list(df35.loc[df35['label'] == i].head(20).word))\n",
    "    print('\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "cluster_labels = {0:'gut',1:'biased signaling',2:'hedgehog signaling',3:'olfactory',4:'no_use',5:'dopamine',\n",
    "                 6:'genetic',7:'signaling',8:'immune',9:'neuroscience',10:'genome',11:'not-obvious',\n",
    "                 12:'diabete',13:'inflammation',14:'kiss',15:'visual',16:'pharmaceutical',17:'neural signaling',\n",
    "                 18:'pain',19:'assay',20:'not-obvious',21:'neural inflammation',22:'pharma',23:'no_use',24:''}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "dopamine = df35.loc[df35['label'] == 5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(5514, 2)"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dopamine.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Other clustering methods"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ward_cluster = ward_tree(word_vecs,n_clusters=n_cluster) # not working\n",
    "d = distance.cdist(word_vecs,ward_cluster.cluster_centers_,metric='euclidean')\n",
    "SS = np.sum(np.min(d,axis=1))/len(word_vecs)\n",
    "print('ward_cluster done, loss = {:.3}'.format(SS))\n",
    "print(datetime.now())\n",
    "\n",
    "ward_labels = ward_cluster.labels_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "birch_cluster = Birch(n_clusters=n_cluster).fit_predict(word_vecs) # not working\n",
    "d = distance.cdist(word_vecs,birch_cluster.cluster_centers_,metric='euclidean')\n",
    "SS = np.sum(np.min(d,axis=1))/len(word_vecs)\n",
    "print('birch_cluster done, loss = {:.3}'.format(SS))\n",
    "print(datetime.now())\n",
    "\n",
    "birch_labels = birch_cluster.labels_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "agg_cluster = AgglomerativeClustering(n_clusters=n_cluster).fit(word_vecs) # not working\n",
    "d = distance.cdist(word_vecs,agg_cluster.cluster_centers_,metric='euclidean')\n",
    "SS = np.sum(np.min(d,axis=1))/len(word_vecs)\n",
    "print('AgglomerativeClustering done, loss = {:.3}'.format(SS))\n",
    "print(datetime.now())\n",
    "\n",
    "agg_labels = agg_cluster.labels_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
